outputs/jira_test.go

```diff
@@ -0,0 +1,365 @@
+			jiraAPI:       &JiraAPI{},
+			metaProject:   &jira.MetaProject{IssueTypes: []*jira.MetaIssueType{{Name: "Task"}, {Name: "Bug"}}},
+			wantIssueType: "Task",
+		},
+		{
+			name:          "happy path (empty issueType, jira doesn't have 'Task' field)",
+			jiraAPI:       &JiraAPI{},
+			metaProject:   &jira.MetaProject{IssueTypes: []*jira.MetaIssueType{{Name: "Story"}, {Name: "Bug"}}},
+			wantIssueType: "Story",
+		},
+		{
+			name:          "happy path (fill priority, jira have 'Bug' field)",
+			jiraAPI:       &JiraAPI{Issuetype: "Bug"},
+			metaProject:   &jira.MetaProject{IssueTypes: []*jira.MetaIssueType{{Name: "Task"}, {Name: "Bug"}}},
+			wantIssueType: "Bug",
+		},
+		{
+			name:        "bad path (fill priority, jira doesn't have 'Bug' field)",
+			jiraAPI:     &JiraAPI{Issuetype: "Bug"},
+			metaProject: &jira.MetaProject{IssueTypes: []*jira.MetaIssueType{{Name: "Task"}, {Name: "Story"}}},
+			wantError:   "project \"\" don't have issueType \"Bug\"",
+		},
+		{
+			name:        "bad path (metaIssueType have empty IssueTypes)",
+			jiraAPI:     &JiraAPI{Priority: ""},
+			metaProject: &jira.MetaProject{IssueTypes: []*jira.MetaIssueType{}},
+			wantError:   "project \"\" don't have issueTypes",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := createIssueType(test.jiraAPI, test.metaProject)
+
+			if test.wantError != "" {
+				require.NotNil(t, err)
+				assert.Contains(t, err.Error(), test.wantError)
+			} else {
+				require.Equal(t, test.wantIssueType, test.jiraAPI.Issuetype)
+			}
+		})
+	}
+}
+
+func TestCreateMetaProject(t *testing.T) {
+	tests := []struct {
+		name               string
+		metaInfo           interface{}
+		wantMetaProjectKey string
+		wantError          string
+	}{
+		{
+			name: "happy path",
+			metaInfo: jira.CreateMetaInfo{
+				Projects: []*jira.MetaProject{
+					{Key: "test"},
+					{Key: "debug"},
+				},
+			},
+			wantMetaProjectKey: "debug",
+		},
+		{
+			name:      "sad path (jira return error)",
+			metaInfo:  "bad struct",
+			wantError: "failed to get create meta : json: cannot unmarshal string into Go value of type jira.CreateMetaInfo",
+		},
+		{
+			name: "sad path (project not found)",
+			metaInfo: jira.CreateMetaInfo{
+				Projects: []*jira.MetaProject{
+					{Key: "test"},
+					{Key: "debug"},
+				},
+			},
+			wantMetaProjectKey: "bad project",
+			wantError:          "could not find project with key bad project",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(200)
+				metaInfoJson, err := json.Marshal(test.metaInfo)
+				if err != nil {
+					t.Errorf("Unexpected marshal metaInfo error: %v", err)
+				}
+				_, err = w.Write(metaInfoJson)
+				if err != nil {
+					t.Errorf("Unexpected write response error: %v", err)
+				}
```

Comment:
This error handling is not necessary.

```suggestion
				w.WriteHeader(200)
				metaInfoJson, _ := json.Marshal(test.metaInfo)
				_, _ = w.Write(metaInfoJson)
```

Reply:
Done